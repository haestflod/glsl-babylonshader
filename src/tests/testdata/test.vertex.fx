attribute vec3 aPosition;
attribute vec2 aUV;

varying vec3 vPosition;
varying vec2 vUV;

uniform vec4 uForm; // after variable linecomment
		
// Main function!
void main(void)
{
#ifdef PREPROCESSOR
	vec4 testcolor = vec4(1.0, 1.0, 1.0, 1.0);
#endif


	/****** block comment */
	if (vUV != 1.0 ) {
		/****** another block comment ******/
		float test = 1.0;
	} else {
		float test2 = 2.0;
	}

	float x = 1.0; /* comment *//* 2ndComment */ float y = 1.0; /* ThirdComment*/ // float z = 1.0;
	// float z = 2.0;
	/*


	Multi line comment
	float z = 3.0;
	*/ float z = 4.0; // After line comment
	

	

	// Good thing we're not validating output then this would crash :p
	gl_Position = vPos;
}
