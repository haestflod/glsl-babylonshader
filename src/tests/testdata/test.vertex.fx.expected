BABYLON.Effect.ShadersStore["TestVertexShader"]=
"attribute vec3 aPosition;"+
"attribute vec2 aUV;"+
"varying vec3 vPosition;"+
"varying vec2 vUV;"+
"uniform vec4 uForm;"+
"void main(void)"+
"{"+
"#ifdef PREPROCESSOR"+
"vec4 testcolor = vec4(1.0, 1.0, 1.0, 1.0);"+
"#endif"+
"if (vUV != 1.0 ) {"+
"float test = 1.0;"+
"} else {"+
"float test2 = 2.0;"+
"}"+
"float x = 1.0; float y = 1.0;"+
"float z = 4.0;"+
"gl_Position = vPos;"+
"}";